File: Domasna 3/backend/backend/src/main/java/ukim/finki/dians/backend/model/Review.java
package ukim.finki.dians.backend.model;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;

@Data
@Entity
@NoArgsConstructor
public class Review {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private Integer grade;
    @Column(length = 400)
    private String comment;
    private LocalDate date;
    @ManyToOne
    private Visitor visitor;

    public Review(Integer grade, String comment, Visitor visitor) {
        this.grade = grade;
        this.comment = comment;
        this.visitor = visitor;
        date=LocalDate.now();
    }
}

File: Domasna 4/frontend/frontend/src/app/Review.ts
import {Visitor} from "./Visitor";

export class Review {

  id: number;
  grade: number;
  comment: string;
  date: Date;
  visitor: Visitor;

  constructor(id: number, grade: number, comment: string, date: Date, visitor: Visitor) {

    this.id = id;
    this.grade = grade;
    this.comment = comment;
    this.date = date;
    this.visitor = visitor;
  }

}
